### DISCUSSION TITLE
What do I do after learning the basics of the Python language? Is there a source for making projects or exercises?

### DISCUSSION THREAD

**Discussion by 1HHHHHHHHHH1** [community/community#29724] [Other Feature Feedback, Questions, & Ideas]
What do I do after learning the basics of the Python language? Is there a source for making projects or exercises?

**Comment by airtower-luna** [comment]
It really depends on what you want to get into next. Make some tool you'd like? Web service development? Data science/visualization? There are so many options, and that was nowhere near an exhaustive list. If you have a direction you want to go, look for something in that direction (or ask again). :slightly_smiling_face: 

If you want some general practice, look for coding challenges or puzzles. Here's [one I recently had fun with](https://2022.coding-challenge.festo.com/), but you can find a lot more online.

**Reply by 1HHHHHHHHHH1** [reply]
I want to manufacture systems and applications

**Reply by airtower-luna** [reply]
That's way too unspecific to give hints, I'm afraid. Any programmer who isn't programming purely for art does that. And even then you could call it an artistic application, if you want. :sweat_smile:

**Comment by mmqeth** [comment]
I really recommend building discord bots in pycord, it's really helpful for strengthening skills and generating products. In fact, it is even easier to code with nodejs. If you are going to try my suggestion, please be careful not to use discord.py, it has been officially deprecated.

**Comment by alexwholland** [comment]
I would suggest browsing YouTube for some project ideas. I say YouTube because it allows you to discover and visualize a ton of resources and project ideas in a short amount of time.
